contents/node/nodejs.md

```diff
@@ -0,0 +1,203 @@
+# Introduction to Node.js
+
+Author: Rachael Sim
+
+# Overview
+
+*Node.js* is a JavaScript runtime built on Chrome’s V8 JavaScript engine. Node.js uses an event-driven, non-blocking I/O model that makes it lightweight and efficient.
+
+### Event-driven
+
+When an event happens e.g. an I/O event is complete, the event handler/callback function will be enqueued in a queue where it will be subsequently scheduled to run by the [event loop](https://medium.com/the-node-js-collection/what-you-should-know-to-really-understand-the-node-js-event-loop-and-its-metrics-c4907b19da4c).
+
+### Non-Blocking
+
+When file and other I/O operations requests are made in Python or Java, they are blocking - subsequent requests (in the same thread) have to be enqueued and can only be processed after the current operation completes. Meanwhile, the program remains idle. In contrast, Node.js has an event loop which delegates the actual tasks to other systems (e.g. file system and databases) and it is almost never blocked after delegation. While an I/O operation is incomplete, the event loop can still process subsequent requests.
+
+### Benefits
+
+* Node.js is fast for I/O intensive tasks due to the non-blocking model.
+* As Node is single-threaded, we can avoid thread overheads and synchronization problems including deadlocks and race conditions.
+* Good support - largest ecosystem of open source libraries in the world
+* Same language (JavaScript) for front and back-end development. This makes it possible to share and reuse code between the front-end and back-end. Teams can be more efficient and less reliant on documentation as they can read the code directly. The team can also become more cross-functional as developers can contribute to both development.
+
+### Use cases
+
+To find out more about when you should or should not use Node.js, checkout these articles by [netguru](https://www.netguru.co/blog/use-node-js-backend) and [Node.js foundation](https://medium.com/the-node-js-collection/why-the-hell-would-you-use-node-js-4b053b94ab8e).
```

Comment:
Can you give at least a few examples and point the reader to other articles to find more. Simply pointing to other articles feels like cheating :-p